### ISSUE TITLE
Why MES loss is nan in training?

### ISSUE THREAD

**Issue by bestl1fe** [chairc/Integrated-Design-Diffusion-Model#60]
When I trained 120 size images to epoch 30, **nan** appeared and all images turned black. I didn't save the model for each training, so I had to retrain my model.

I think we should add a way to prevent nan.

![image](https://github.com/chairc/Integrated-Design-Diffusion-Model/assets/166411263/6bfc6ac4-20a7-42aa-8741-37befc03c5a6)

**Comment by chairc** [comment]
Good work. Could you submit a pr for this issue?

**Comment by chairc** [comment]
The possible reasons for this problem may include an excessively high learning rate, unstable operations in the loss function, issues with the activation function, etc. **Of course, future versions will be updated to include a check for NaN values.**

**Comment by chairc** [comment]
We submit a function for NaN detection #89
